packages/firestore/src/local/simple_db.ts

```diff
@@ -406,6 +406,18 @@ export interface IterateOptions {
   reverse?: boolean;
 }
 
+/** An error that wraps exceptions that thrown during IndexedDB execution. */
+export class IndexedDbTransactionError extends FirestoreError {
+  name = 'IndexedDbTransactionError';
+
+  constructor(cause: Error) {
+    super(
+      Code.UNKNOWN,
+      'IndexedDB transaction failed with environment error: ' + cause
```

Comment:
Drop "with environment error". "IndexedDB transaction failed" is enough to indicate what's failing and the cause will carry the details.